splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -441,23 +442,24 @@ long now() {
     }
 
     private static class LazyInitSpanExporter implements SpanExporter {
-        private volatile SpanExporter delegate;
+        @Nullable private volatile SpanExporter delegate;
         private final Supplier<SpanExporter> s;
 
         public LazyInitSpanExporter(Supplier<SpanExporter> s) {
             this.s = s;
         }
 
         private SpanExporter getDelegate() {
-            if (delegate != null) {
-                return delegate;
-            }
-            synchronized (this) {
-                if (delegate == null) {
-                    delegate = s.get();
+            SpanExporter d = delegate;
+            if (d == null) {
+                synchronized (this) {
+                    d = delegate;
+                    if (d == null) {
+                        delegate = d = s.get();
+                    }
```

Comment:
This construct always gives me pause because it was broken for so long...and it's fairly complicated to reason about. Since the volatile is required for it to be "correct" I wonder if just using a lock or synchronizing the method isn't just simpler/better. I won't lose sleep over it tho.

Reply:
It's worth noting that this whole thing will be replaced by just one word `lazy` when we finally migrate to Kotlin.

> I wonder if just using a lock or synchronizing the method isn't just simpler/better.

It probably is a lot simpler, but I don't know enough about the Android JVM/Dalvik/whatever to say how slower this will be; a volatile field access is supposed to be just one instruction, no idea about the `sychronized` monitor.